### PR TITLE
feat: update dependencies and add dart_code_linter

### PR DESCRIPTION
**Pull request by fedecor9**
# :star: Feature

#### :pencil2: Description:

Updated package dependencies and flutter version.
Fixed deprecated classes from latest flutter version and added dart_code_linter as code analyzer

### CODE REVIEW

File: catalog/lib/theme/app_color_scheme.dart [hunk 1]
```diff
@@ -88,8 +86,8 @@ class AppColorScheme extends ColorScheme {
             },
           ),
           onError: Colors.black,
-          onBackground: const Color(0xffadadad),
-          background: const Color(0xfff7fafd),
+          onSurface: const Color(0xffadadad),
+          surface: const Color(0xfff7fafd),
```
Comment (f7deleon) [inline catalog/lib/theme/app_color_scheme.dart:90]: do we have to delete background and onBackground?

**Review by f7deleon — APPROVED** [review]
Nice 🚀  weird that now we have to use double quotes for yaml

File: analysis_options.yaml [hunk 2]
```diff
@@ -14,8 +14,6 @@ linter:
     - avoid_relative_lib_imports
     - avoid_renaming_method_parameters
     - avoid_return_types_on_setters
-    - avoid_returning_null
-    - avoid_returning_null_for_future
```
[the remark this answers is not in the document]
Comment (fedecor9) [reply analysis_options.yaml:18]: NOTE: This rule is removed in Dart 3.3.0; it is no longer functional.

Both rules say this in the docs
https://dart.dev/tools/linter-rules/avoid_returning_null

File: catalog/lib/theme/app_color_scheme.dart (hunk 1 above)
Comment (fedecor9) [reply catalog/lib/theme/app_color_scheme.dart:90]: `onBackground` and `background` is deprecated in the latest version
https://api.flutter.dev/flutter/material/ColorScheme/onBackground.html

**Review by mirland — APPROVED** [review]

**Pull request merged**